src/goto-instrument/contracts/dynamic-frames/dfcc.cpp

```diff
@@ -47,6 +47,66 @@ Author: Remi Delmas, delmarsd@amazon.com
 
 #include "dfcc_lift_memory_predicates.h"
 
+invalid_function_contract_pair_exceptiont::
+  invalid_function_contract_pair_exceptiont(
+    std::string reason,
+    std::string correct_format)
+  : cprover_exception_baset(std::move(reason)),
+    correct_format(std::move(correct_format))
+{
+}
+
+std::string invalid_function_contract_pair_exceptiont::what() const
+{
+  std::string res;
+
+  res += "Invalid function-contract mapping";
+  res += "\nReason: " + reason;
+
+  if(!correct_format.empty())
+  {
+    res += "\nFormat: " + correct_format;
+  }
+
+  return res;
+}
+
+static std::pair<irep_idt, irep_idt>
+parse_function_contract_pair(const irep_idt &cli_flag)
+{
+  auto const correct_format_message =
+    "the format for function and contract pairs is "
+    "`<function_name>[/<contract_name>]`";
+
+  std::string cli_flag_str = id2string(cli_flag);
+  auto const function_name_end = cli_flag_str.find('/');
+  if(function_name_end == std::string::npos)
+  {
+    // <function_name> -> (<function_name>, <function_name>)
+    return std::make_pair(cli_flag, cli_flag);
+  }
+
+  // <function_name>/<contract_name> -> (<function_name>, <contract_name>)
+  std::string function_name = cli_flag_str.substr(0, function_name_end);
+  std::string contract_name = cli_flag_str.substr(function_name_end + 1);
```

Comment:
You might use `split_string(cli_flag_str, '/')` and then inspect the resulting vector.

Reply:
fixed